tests/test_ip_info.py

```diff
@@ -2,66 +2,571 @@
+    mocker.patch.object(ip_info, "get_vendor_offline", return_value=None)
+    mocker.patch.object(
+        ip_info, "get_vendor_online", return_value=online_vendor
+    )
+
+    result = ip_info.get_vendor(mac_addr, profileid)
+
+    assert result == {"MAC": mac_addr, "Vendor": online_vendor}
+    (mock_db.get_mac_vendor_from_profile.assert_called_once_with(profileid))
+    (ip_info.get_vendor_offline.assert_called_once_with(mac_addr, profileid))
+    (ip_info.get_vendor_online.assert_called_once_with(mac_addr))
+    mock_db.set_mac_vendor_to_profile.assert_called_once_with(
+        profileid, mac_addr, online_vendor
+    )
+
+
+def test_get_vendor_not_found(mocker, mock_db):
+    ip_info = ModuleFactory().create_ip_info_obj(mock_db)
+    mac_addr = "08:00:27:7f:09:e4"
+    profileid = "profile_10.0.2.18"
+
+    mock_db.get_mac_vendor_from_profile.return_value = None
+    mocker.patch.object(ip_info, "get_vendor_offline", return_value=None)
+    mocker.patch.object(ip_info, "get_vendor_online", return_value=None)
+
+    result = ip_info.get_vendor(mac_addr, profileid)
+
+    assert result == {"MAC": mac_addr, "Vendor": "Unknown"}
+    (mock_db.get_mac_vendor_from_profile.assert_called_once_with(profileid))
+    (ip_info.get_vendor_offline.assert_called_once_with(mac_addr, profileid))
+    ip_info.get_vendor_online.assert_called_once_with(mac_addr)
+    mock_db.set_mac_vendor_to_profile.assert_not_called()
+
+
+def test_get_vendor_broadcast_mac(mocker, mock_db):
+    ip_info = ModuleFactory().create_ip_info_obj(mock_db)
+    mac_addr = "ff:ff:ff:ff:ff:ff"
+    profileid = "profile_10.0.2.19"
+
+    result = ip_info.get_vendor(mac_addr, profileid)
+
+    assert result is False
+    mock_db.get_mac_vendor_from_profile.assert_not_called()
+    mocker.patch.object(ip_info, "get_vendor_offline").assert_not_called()
+    mocker.patch.object(ip_info, "get_vendor_online").assert_not_called()
+    mock_db.set_mac_vendor_to_profile.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "domain, get_domain_data_return_value, "
+    "whois_creation_date, expected_age",
+    [
+        # Testcase 1: Returns correct age for a
+        # valid domain with existing data
+        ("google.com", None, None, 100),
+        # Testcase 2: Returns False for a domain
+        # with no creation date in whois
+        ("example.com", None, None, False),
+        # Testcase 3: Returns False for a domain
+        # with an invalid TLD
+        ("example.invalid", None, None, False),
+    ],
+)
+def test_get_age(
+    mock_db,
+    domain,
+    get_domain_data_return_value,
+    whois_creation_date,
+    expected_age,
+):
+    ip_info = mock_db.create_ip_info_obj()
+    mock_db.get_domain_data.return_value = get_domain_data_return_value
+    with mock.patch("whois.query") as mock_whois:
+        mock_whois.return_value.creation_date = whois_creation_date
+        ip_info.get_age.return_value = expected_age
+        assert ip_info.get_age(domain) == expected_age
```

Comment:
Here you're patching the get_age() then checking for the patch's result. this will never fail if something's wrong with the original function right? i think we need a better way to test this